R/select_nse.R

```diff
@@ -0,0 +1,458 @@
+# Code adapted from {poorman} by Nathan Eastwood [License: MIT]
+# https://github.com/nathaneastwood/poorman/blob/master/R/select_positions.R
+
+.select_nse <- function(select, data, exclude, ignore_case, regex = FALSE,
+                        verbose = FALSE) {
+  .check_data(data)
+  columns <- colnames(data)
+
+  # avoid conflicts
+  conflicting_packages <- .conflicting_packages("poorman")
+  on.exit(.attach_packages(conflicting_packages))
+
+  expr_select <- substitute(select, env = parent.frame(1L))
+  expr_exclude <- substitute(exclude, env = parent.frame(1L))
+
+  # when exclude is not an argument called from the function (e.g data_to_long),
+  # do not consider "exclude" as a symbol
+  if (deparse(expr_exclude) == "exclude" && is.null(substitute(exclude))) {
+    expr_exclude <- NULL
+  }
+
+  # directly return all names if select == exclude == NULL
+  if (is.null(expr_select) && is.null(expr_exclude)) {
+    return(columns)
+  }
+
+  # get the position of columns that are selected or excluded
+  selected <- .eval_expr(expr_select, data,
+    ignore_case = ignore_case,
+    regex = regex, verbose
+  )
+  excluded <- .eval_expr(expr_exclude, data,
+    ignore_case = ignore_case,
+    regex = regex, verbose
+  )
```

Comment:
```suggestion
  selected <- .eval_expr(
    expr_select, 
    data,
    ignore_case = ignore_case,
    regex = regex, verbose
  )
  
  excluded <- .eval_expr(
    expr_exclude,
    data,
    ignore_case = ignore_case,
    regex = regex, verbose
  )
```

Reply:
Can you please also apply one-argument-per-line formatting to the rest of the file (especially the internal select helper functions)?